src/xcfun.cpp

```diff
@@ -729,3 +729,102 @@ int xc_is_gga(xc_functional fun) { return (fun->depends & XC_GRADIENT); }
 int xc_is_metagga(xc_functional fun) {
   return (fun->depends & (XC_LAPLACIAN | XC_KINETIC));
 }
+
+int xc_user_eval_setup(xc_functional fun,
+                       const int order,
+                       const unsigned int func_type, // LDA (0), GGA (1), metaGGA (2), taylor (3)
+                       const unsigned int dens_type,  // A (0), N (1), A_B (2), N_S (3)
+                       const unsigned int mode_type,  // same as the enum list
+                       const unsigned int laplacian,  // 0/1 laplacian no/yes 
+                       const unsigned int kinetic,    // 0/1 kinetic energy no/yes
+                       const unsigned int current,    // 0/1 current density no/yes
+                       const unsigned int explicit_derivatives) {   // 0/1 gamma vs explicit partial derivatives 
+
+    if (func_type > 3 || dens_type > 3 || mode_type > 3 || laplacian > 1 || kinetic > 1 || current > 1 || explicit_derivatives > 1) {
+        xcint_die("xc_user_eval_setup: invalid input",-1);
+    }
+    
+
+    enum xc_vars vars = XC_VARS_UNSET; 
+    enum xc_mode mode = XC_MODE_UNSET;
+    
+    switch (mode_type){
+    case(1): mode = XC_PARTIAL_DERIVATIVES; break;
+    case(2): mode = XC_POTENTIAL;           break;
+    case(3): mode = XC_CONTRACTED;          break;
+    default:
+        xcint_die("xc_user_eval_setup: Invalid mode", mode_type);
+    }
+
+
```

Comment:
Before running `clang-format`, enclose this comment block in a `// clang-format off`/`clang-format on` block, so that it doesn't get all mangled.